components/automate-ui/src/app/components/pending-edits-bar/pending-edits-bar.component.spec.ts

```diff
@@ -219,6 +171,156 @@ describe('PendingEditsBarComponent', () => {
+      expect(component.isBarHidden).toEqual(true);
+    });
+
+    it('is hidden if update is cancelled', () => {
+      store.dispatch(
+        new GetProjectsSuccess({ projects: [genProject('uuid-99', 'RULES_APPLIED')] }));
+      component.confirmApplyStart();
+      store.dispatch(new GetApplyRulesStatusSuccess(
+        genState(ApplyRulesStatusState.NotRunning, false, true)));
+      component.updateDisplay();
+      expect(component.isBarHidden).toEqual(true);
+    });
+
+    it('is hidden if update is cancelled but update is still running', () => {
+      store.dispatch(
+        new GetProjectsSuccess({ projects: [genProject('uuid-99', 'RULES_APPLIED')] }));
+      component.confirmApplyStart();
+      store.dispatch(new GetApplyRulesStatusSuccess(
+        genState(ApplyRulesStatusState.Running, false, true)));
+      component.updateDisplay();
+      expect(component.isBarHidden).toEqual(true);
+    });
+  });
+});
+
+
+function configureWith(perms: IndexedEntities<UserPermEntity>): void {
+  TestBed.configureTestingModule({
+    declarations: [
+      MockComponent({
+        selector: 'chef-toolbar',
+        template: '<ng-content></ng-content>'
+      }),
+      MockComponent({
+        selector: 'app-authorized',
+        inputs: ['allOf', 'not'],
+        template: '<ng-content></ng-content>'
+      }),
+      MockComponent({
+        selector: 'app-message-modal',
+        inputs: ['visible'],
+        outputs: ['close']
+      }),
+      MockComponent({
+        selector: 'app-confirm-apply-start-modal',
+        inputs: ['visible'],
+        outputs: ['confirm', 'cancel']
+      }),
+      MockComponent({
+        selector: 'app-confirm-apply-stop-modal',
+        inputs: ['visible', 'applyRulesStatus', 'stopRulesInProgress'],
+        outputs: ['confirm', 'cancel']
+      }),
+      MockComponent({ selector: 'mat-progress-bar', inputs: ['mode', 'value', 'bufferValue'] }),
+      MockComponent({ selector: 'mat-select' }),
+      MockComponent({ selector: 'mat-option' }),
+      MockComponent({ selector: 'chef-button', inputs: ['disabled'] }),
+      MockComponent({ selector: 'chef-heading' }),
+      MockComponent({ selector: 'chef-loading-spinner' }),
+      MockComponent({ selector: 'chef-page-header' }),
+      MockComponent({ selector: 'chef-subheading' }),
+      MockComponent({ selector: 'chef-table-new' }),
+      MockComponent({ selector: 'chef-table-header' }),
+      MockComponent({ selector: 'chef-table-body' }),
+      MockComponent({ selector: 'chef-table-row' }),
+      MockComponent({ selector: 'chef-table-header-cell' }),
+      MockComponent({ selector: 'chef-table-cell' }),
+      PendingEditsBarComponent
+    ],
+    imports: [
+      ReactiveFormsModule,
+      RouterTestingModule,
+      ChefPipesModule,
+      StoreModule.forRoot({
+        ...ngrxReducers,
+          userperms: () => <PermEntityState>{
+            status: Status.loadingSuccess,
+            byId: perms
+          }
```

Comment:
... with this one change: allowing the `userperms` portion of the ngrx store to be configured by the passed-in `perms` parameter.